src/_pytest/capture.py

```diff
@@ -692,6 +704,39 @@ def snap(self):
         return res
 
 
+class OrderedCapture(SysCapture):
+    """This class uses deque to keep streams order."""
+    streams = collections.deque()
+
+    def _get_writer_class(self):
+        return OrderedWriter(self.fd)
+
+    def snap(self):
+        res = self.tmpfile.getvalue()
+        if self.name == "stderr":
+            # both streams are being read one after another, while stderr is last - it will clear the queue
+            self.streams.clear()
+        return res
+
+
+class OrderedWriter:
+    encoding = sys.getdefaultencoding()
+
+    def __init__(self, fd):
```

Comment:
```suggestion
    def __init__(self, fd: int) -> None:
```
I.e. please add type annotations where you can.

Reply:
Sure, I'll do that. I wasn't sure if that's allowed.